packages/design-system/src/components/tree/tree-node.tsx

```diff
@@ -282,6 +282,16 @@ export const TreeItemBody = <Data extends { id: string }>({
   const isDragging = dropTargetItemId !== undefined;
   const isDropTarget = dropTargetItemId === itemData.id;
 
+  // Scroll the selected button into view when selected from canvas.
+  useEffect(() => {
+    if (isSelected && isDragging === false && isDropTarget === false) {
+      itemButtonRef.current?.scrollIntoView({
+        behavior: "smooth",
+        block: "nearest",
+      });
+    }
+  }, [isSelected, isDragging, isDropTarget]);
```

Comment:
Seems like this will cause scroll not only when it becomes selected, but also when `isDragging` or `isDropTarget` becomes `false`.

Not sure this is a real scenario, but I would use refs to make `isSelected` the only dependency just in case

Reply:
yep, done